### PR TITLE
[CAZB-2850] Account Maintenance - Update password page

### PR DESCRIPTION
**Pull request by pkalamarski**
## Motivation and Context

Parent: https://eaflood.atlassian.net/browse/CAZB-2637
https://eaflood.atlassian.net/browse/CAZB-2850

## Description

Added update password page.

## How Has This Been Tested?

Manually, rspec, cucumber

## Screenshots (if appropriate):
image
image
image

## Checklist:

- [x] It contains only changes required by issue (does not contain other PR)
- [x] Includes link to an issue (if apply)
- [x] I have added tests to cover my changes.

### CODE REVIEW

**Comment by pkalamarski** [comment]
Implemented in #581

File: features/account_details/passwords.feature [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  Scenario: Password update process as non primary user
+    Given I navigate to a Dashboard page with 'MANAGE_VEHICLES' permission
+      Then I want to see my account details
+      And I want to change my password
+      And I should be on account details update password page
+    Then I press 'Exit without saving' link
+      And I should be on non primary user account details page
+      And I want to change my password
+    When I fill in correct old and new password and press 'Save changes'
+      Then I should be on non primary user account details page
```
Comment (likeuwill) [inline features/account_details/passwords.feature:39]: empty line missing

File: app/views/account_details/passwords/edit.html.haml [hunk 2]
```diff
@@ -1,7 +1,62 @@
 - title_and_header = 'Update password'
 - content_for(:title, title_and_header)
+- return_url = current_user.owner ? primary_users_account_details_path : non_primary_users_account_details_path
```
Comment (likeuwill) [inline app/views/account_details/passwords/edit.html.haml:3]: can u move this logic to `edit` method?

Comment (likeuwill) [reply app/views/account_details/passwords/edit.html.haml:3]: we need try to escape using such a logic in views when we have space in controllers :)

File: app/controllers/account_details/passwords_controller.rb [hunk 3]
```diff
@@ -9,28 +9,49 @@ module AccountDetails
...
-      end
+    # Renders :edit with assigned errors
+    def rerender_edit(errors)
+      @errors = errors
+      render :edit
+    end
+
+    # Redirects to a proper account management page
+    def redirect_to_account_management
+      url = current_user.owner ? primary_users_account_details_path : non_primary_users_account_details_path
```
Comment (likeuwill) [inline app/controllers/account_details/passwords_controller.rb:53]: u need this method in two places, so u can add a new method and just use it in `redirect_to url` and in `@back_button_url`

**Review by likeuwill — CHANGES_REQUESTED** [review]

**Review by likeuwill — APPROVED** [review]

**Pull request merged**